### PR TITLE
fix: Issues with segment

### PR DESCRIPTION
**Pull request by benjackwhite**
* Corrects identify to work with anonymous users as well via `$set` (following this [comment](https://github.com/PostHog/posthog-js/pull/584#pullrequestreview-1370611979))
* We weren't putting the $set in the properties 🤦  - now its fixed
* Adds tests which is how I found all these issues out 😉

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -49,32 +49,32 @@ async function onTrack(event, settings) {
...
         }
-        event.properties["$groups"] = {"segment_group": event.context.groupId}
+        event.properties['$groups'] = { segment_group: event.context.groupId }
     }
-  
 
     const res = await fetch(endpoint, {
         body: JSON.stringify({
-            ...event,
+            timestamp: event.timestamp,
```
Comment (timgl) [inline index.js:70]: Why this change?

Comment (benjackwhite) [reply index.js:70]: We are sending a tonne of stuff that we don't need which is confusing to maintain so I opted to just send the properties we actually support. Clearer and easier to reason on in the future

File: index.js [hunk 2]
```diff
@@ -92,21 +92,33 @@ async function onTrack(event, settings) {
...
-        },
-        settings
-    )
+    properties = {
+        ...(properties || {}),
+        $set: event.traits,
+    }
+
+    if (event.userId) {
+        properties['$anon_distinct_id'] = event.anonymousId
```
Comment (timgl) [inline index.js:102]: if `event.anonymousId` is empty this will be undefined, should it be an if instead?

Comment (benjackwhite) [reply index.js:102]: I got the impression that both will be set but thats a good point. I'll change it

**Review by timgl — APPROVED** [review]
